### PR TITLE
Clustering added

### LINKED ISSUE #46: Clustering
# Prerequisites

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

# Expected Behavior

Sever should run in cluster mode to run multiple instances of the server in fork mode to make it faster.

# Current Behavior

Server is currently running in a non cluster mode.

## Context

[Reason](https://www.w3schools.com/nodejs/ref_cluster.asp#:~:text=same%20server%20port.-,Node.,thread%2C%20to%20handle%20the%20load.)

[Module](https://www.npmjs.com/package/cluster)

I would like to work on this issue as a gssoc 21 participant.

### PR DESCRIPTION
**Pull request by Shubham-Kumar-2000**
# Description

Renamed the index.js file as app.js and exported a function start from it.
Created a new file index.js to support clustering and starting server from each worker.

Fixes #46 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Start in cluster mode start many servers each on a different CPU core.
- [x] Start in non cluster mode starts only one server.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

@Manvityagi  please look into it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,8 +21,10 @@
   "homepage": "https://github.com/Manvityagi/Opportunity-Calendar-Backend#readme",
   "dependencies": {
     "body-parser": "^1.19.0",
+    "cluster": "^0.7.7",
     "cors": "^2.8.5",
     "dotenv": "^8.2.0",
+    "esm": "^3.2.25",
```
Comment (abdus) [inline package.json]: why do we need this package?

**Review by abdus — COMMENTED** [review]
there's a `.prettierrc` file in this repository. please format your code with prettier.

File: package.json (hunk 1 above)
Comment (Shubham-Kumar-2000) [reply package.json]: Sorry removed it.
It was added by mistake.

**Comment by Shubham-Kumar-2000** [comment]
Formatted with prettier @abdus

**Review by abdus — APPROVED** [review]
Looks good to me.

**Review by Manvityagi — APPROVED** [review]

**Comment by Shubham-Kumar-2000** [comment]
Can you explain how is it a level 1 PR??
according to this
image

it was not a major documention
I contributed for a level 3 but at the very least it comes under level 2

@Manvityagi

**Comment by Manvityagi** [comment]
> Can you explain how is it a level 1 PR??
> according to this
> image
> 
> it was not a major documention
> I contributed for a level 3 but at the very least it comes under level 2
> 
> @Manvityagi

Changed to Level-2.

**Pull request merged**